### PR TITLE
Drop py2 support

### PR DESCRIPTION
**Pull request by aodag**

### CODE REVIEW

File: pandas_td/test_td.py [hunk 1]
```diff
@@ -102,30 +98,33 @@ def tearDown(self):
         os.environ.clear()
         os.environ.update(self._environ)
 
-    @raises(ValueError)
     def test_error_without_parameters(self):
-        Connection()
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: Personally, it's explicit to use `assertRaises` like:

```py
with self.assertRaises(ValueError):
    Connection()
```

File: pandas_td/test_td.py [hunk 2]
```diff
@@ -189,18 +188,21 @@ def test_execute_ok(self):
...
 
-    @raises(RuntimeError)
     def test_execute_error(self):
         # mock
         job = MockJob('error')
         self.connection.client.query = MagicMock(return_value=job)
         # test
         engine = QueryEngine(self.connection, 'test_db')
-        r = engine.execute('select 1')
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 3]
```diff
@@ -322,9 +324,12 @@ def setUp(self):
     def assert_query(self, query):
         self.connection.client.query.assert_called_with('test_db', "-- read_td_table('test_table')" + query, type='presto')
 
-    @raises(ValueError)
     def test_invalid_time_range(self):
-        read_td_table('test_table', self.engine, time_range=(1.0, 2.0))
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 4]
```diff
@@ -387,28 +392,39 @@ def setUp(self):
         self.connection.client = self.mock_client()
         self.frame = pd.DataFrame([[1,2],[3,4]], columns=['x', 'y'])
 
-    @raises(ValueError)
     def test_invalid_table_name(self):
-        to_td(self.frame, 'invalid', self.connection)
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 5]
```diff
@@ -387,28 +392,39 @@ def setUp(self):
...
+        except ValueError:
+            pass
 
-    @raises(TypeError)
     def test_datetime_is_not_supported(self):
         client = self.connection.client
         # test
         frame = pd.DataFrame({'timestamp': [datetime.datetime(2000,1,1)]})
-        to_td(frame, 'test_db.test_table', self.connection)
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 6]
```diff
@@ -387,28 +392,39 @@ def setUp(self):
...
+            assert False
+        except TypeError:
+            pass
 
     # if_exists
 
-    @raises(ValueError)
     def test_invalid_if_exists(self):
-        to_td(self.frame, 'test_db.test_table', self.connection, if_exists='invalid')
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 7]
```diff
@@ -387,28 +392,39 @@ def setUp(self):
...
+            assert False
+        except ValueError:
+            pass
 
-    @raises(RuntimeError)
     def test_fail_if_exists(self):
         client = self.connection.client
         client.table = MagicMock()
-        to_td(self.frame, 'test_db.test_table', self.connection)
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 8]
```diff
@@ -436,34 +452,39 @@ def test_append_if_exists(self):
 
     # time_col
 
-    @raises(ValueError)
     def test_error_time_col_and_time_index(self):
-        _convert_time_column(self.frame, time_col='x', time_index=0)
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 9]
```diff
@@ -436,34 +452,39 @@ def test_append_if_exists(self):
...
+            _convert_time_column(self.frame, time_col='x', time_index=0)
+            assert False
+        except ValueError:
+            pass
 
-    @raises(ValueError)
     def test_error_time_column_already_exists(self):
         f1 = pd.DataFrame([[0, 'a', 1], [0, 'b', 2]], columns=['time', 'x', 'y'])
-        f2 = _convert_time_column(f1)
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 10]
```diff
@@ -473,114 +494,132 @@ def test_time_col_by_unixtime_substituted(self):
...
+        assert list(f2['time'].values) == [978307200, 978307200]
 
     # time_index
 
-    @raises(TypeError)
     def test_invalid_arg_time_index(self):
         date_range = pd.date_range('2015-01-01', periods=2, freq='d')
         f1 = pd.DataFrame([['a', 1], ['b', 2]], columns=['x', 'y'], index=date_range)
-        f2 = _convert_time_column(f1, time_index=True)
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 11]
```diff
@@ -473,114 +494,132 @@ def test_time_col_by_unixtime_substituted(self):
...
+            assert False
+        except TypeError:
+            pass
 
-    @raises(IndexError)
     def test_invalid_level_time_index(self):
         date_range = pd.date_range('2015-01-01', periods=2, freq='d')
         f1 = pd.DataFrame([['a', 1], ['b', 2]], columns=['x', 'y'], index=date_range)
-        f2 = _convert_time_column(f1, time_index=1)
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 12]
```diff
@@ -473,114 +494,132 @@ def test_time_col_by_unixtime_substituted(self):
...
+            f2 = _convert_time_column(f1, time_index=1)
+            assert False
+        except IndexError:
+            pass
 
-    @raises(TypeError)
     def test_invalid_value_time_index(self):
         f1 = pd.DataFrame([['a', 1], ['b', 2]], columns=['x', 'y'])
-        f2 = _convert_time_column(f1, time_index=0)
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 13]
```diff
@@ -473,114 +494,132 @@ def test_time_col_by_unixtime_substituted(self):
...
         f2 = _convert_time_column(f1, time_index=0)
-        eq_(list(f2.columns), ['x', 'y', 'time'])
+        assert list(f2.columns) == ['x', 'y', 'time']
 
-    @raises(IndexError)
     def test_invalid_level_time_index_multi(self):
         date_range = pd.date_range('2015-01-01', periods=2, freq='d')
         f1 = pd.DataFrame([['a', 1], ['b', 2]], columns=['x', 'y'], index=[[0, 1], date_range])
-        f2 = _convert_time_column(f1, time_index=2)
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 14]
```diff
@@ -473,114 +494,132 @@ def test_time_col_by_unixtime_substituted(self):
...
+            assert False
+        except IndexError:
+            pass
 
-    @raises(TypeError)
     def test_invalid_value_time_index_multi(self):
         date_range = pd.date_range('2015-01-01', periods=2, freq='d')
         f1 = pd.DataFrame([['a', 1], ['b', 2]], columns=['x', 'y'], index=[[0, 1], date_range])
-        f2 = _convert_time_column(f1, time_index=0)
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py [hunk 15]
```diff
@@ -473,114 +494,132 @@ def test_time_col_by_unixtime_substituted(self):
...
-        eq_(list(f2.columns), ['x', 'y', 'time'])
+        assert list(f2.columns) == ['x', 'y', 'time']
 
     # index / index_label
 
-    @raises(TypeError)
     def test_invalid_index_type(self):
         f1 = pd.DataFrame([['a', 1], ['b', 2]], columns=['x', 'y'])
-        f2 = _convert_index_column(f1, index=0)
+        try:
```
Comment (chezou) [inline pandas_td/test_td.py]: ditto

File: pandas_td/test_td.py (hunk 1 above)
Comment (aodag) [reply pandas_td/test_td.py]: Because it's written in unittest.

**Pull request merged**